File: tunnel-server/src/main/java/com/tuuzed/tunnel/server/http/HttpRequestInterceptor.java
package com.tuuzed.tunnel.server.http;

import com.tuuzed.tunnel.common.proto.ProtoRequest;
import io.netty.handler.codec.http.HttpRequest;
import org.jetbrains.annotations.NotNull;

import java.net.SocketAddress;

public interface HttpRequestInterceptor {
    HttpRequestInterceptor DEFAULT = new HttpRequestInterceptor() {
        @Override
        public void proceed(
            @NotNull SocketAddress localAddress,
            @NotNull SocketAddress remoteAddress,
            @NotNull ProtoRequest protoRequest,
            @NotNull HttpRequest httpRequest
        ) throws Exception {
        }
    };

    void proceed(
        @NotNull SocketAddress localAddress,
        @NotNull SocketAddress remoteAddress,
        @NotNull ProtoRequest protoRequest,
        @NotNull HttpRequest httpRequest
    ) throws Exception;
}

File: core/server/src/main/kotlin/lighttunnel/server/interceptor/HttpRequestInterceptor.kt
package lighttunnel.server.interceptor

import io.netty.channel.ChannelHandlerContext
import io.netty.handler.codec.http.FullHttpRequest
import io.netty.handler.codec.http.FullHttpResponse
import lighttunnel.proto.TunnelRequest

interface HttpRequestInterceptor {

    fun handleHttpRequest(ctx: ChannelHandlerContext, tunnelRequest: TunnelRequest, httpRequest: FullHttpRequest): FullHttpResponse? = null

}
